### PR TITLE
Implemented Azure blob based catalog service

### PR DESCRIPTION
**Pull request by coderandhiker**

### CODE REVIEW

**Review by pkgw — APPROVED** [review]
LGTM

File: src/WWT.Azure/Catalog/AzureCatalogAccessor.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        /// </summary>
+        /// <param name="catalogEntryName">The name of the catalog entry to fetch, this will be lower cased</param>
+        public async Task<CatalogEntry> GetCatalogEntryAsync(string catalogEntryName, CancellationToken token)
+        {
+            if (catalogEntryName is null) 
+                return null;
+
+            var blob = _container.GetBlobClient(catalogEntryName.ToLower());
+
+            if (blob is null || !blob.Exists())
```
Comment (twsouthwick) [inline src/WWT.Azure/Catalog/AzureCatalogAccessor.cs:30]: there's an async version of this API.

**Review by twsouthwick — APPROVED** [review]
LGTM

File: src/WWT.Azure/Catalog/AzureCatalogAccessor.cs (hunk 1 above)
Comment (coderandhiker) [reply src/WWT.Azure/Catalog/AzureCatalogAccessor.cs:30]: What's the method?  I understood that fetching the client itself was synchronous.  I don't see a GetBlobClientAsync in the SDK

**Pull request merged**